include/openthread-types.h

```diff
@@ -895,16 +895,19 @@ typedef struct otBufferInfo
  * This structure represents an IPv6 network interface unicast address.
  *
  */
-typedef struct otNetifAddress
+OT_TOOL_PACKED_BEGIN
+struct otNetifAddress
 {
     otIp6Address           mAddress;                 ///< The IPv6 unicast address.
     uint32_t               mPreferredLifetime;       ///< The Preferred Lifetime.
     uint32_t               mValidLifetime;           ///< The Valid lifetime.
     uint8_t                mPrefixLength;            ///< The Prefix length.
-    unsigned int           mScopeOverride : 4;       ///< The IPv6 scope of this address.
-    bool                   mScopeOverrideValid : 1;  ///< TRUE if the mScopeOverride value is valid, FALSE othewrise.
+    uint32_t               mScopeOverride : 4;       ///< The IPv6 scope of this address.
+    uint32_t               mScopeOverrideValid : 1;  ///< TRUE if the mScopeOverride value is valid, FALSE othewrise.
     struct otNetifAddress *mNext;                    ///< A pointer to the next network interface address.
-} otNetifAddress;
+} OT_TOOL_PACKED_END;
```

Comment:
This packed macro is going to create un-optimized accessor instructions. Generally, if objects of a type are not crossing a medium between processors, such as a bus or radio, the packed macro should be avoided.  You should be able to achieve your goals without the packed macro anyways.